### PR TITLE
Limit `dbt-tests-adapter` build to only produce a wheel

### PR DESCRIPTION
**Pull request by mikealfare**
### Problem

We can't currently build an source distribution for `dbt-tests-adapter` because of interactions between `hatch`, `pyproject.toml`, and the fact that this is a monorepo. There are also multiple places in the workflow where `dbt-tests-adapter` is not accounted for (e.g. different working directory, lack of hatch commands, etc.).

### Solution

- Update build steps to only build a wheel
- Update the check steps to only validate a wheel
- Update workflows to add option for running tests
- Update workflows to allow for a working directory to move to `./dbt-tests-adapter`
- Remove duplicated set of steps from the `build` workflow

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-adapter/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have run this code in development, and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc.) or this PR has already received feedback and approval from Product or DX

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -59,19 +59,8 @@ jobs:
         uses: ./.github/actions/build-hatch
         with:
           working-dir: "./dbt-tests-adapter/"
-
```
Comment (mikealfare) [inline .github/workflows/build.yml:62]: This whole set of steps were duplicated for some reason.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -59,19 +59,8 @@ jobs:
...
-      - name: Build `dbt-adapters`
-        if: ${{ inputs.package == 'dbt-adapters' }}
-        uses: ./.github/actions/build-hatch
-
-      - name: Build `dbt-tests-adapter`
-        if: ${{ inputs.package == 'dbt-tests-adapter' }}
-        uses: ./.github/actions/build-hatch
-        with:
-          working-dir: "./dbt-tests-adapter/"
+          build-command: "hatch build -t wheel"
```
Comment (mikealfare) [inline .github/workflows/build.yml:62]: Override the default working directory to cd into the `dbt-tests-adapter` build directory.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -73,18 +73,32 @@ defaults:
 
 jobs:
   bump-version-generate-changelog:
-    name: "Bump package version, Generate changelog"
+    name: "Bump package version, Generate changelog - dbt-adapters"
```
Comment (mikealfare) [inline .github/workflows/release.yml:76]: Duplicate step to provide different configuration for `dbt-adapters` and `dbt-tests-adapter`. `dbt-tests-adapter` has no tests, and `hatch` needs to be in `dbt-tests-adapter` to reference the correct `pyproject.toml` and associated commands.

File: .github/workflows/release_prep_hatch.yml [hunk 4]
```diff
@@ -53,6 +53,9 @@ on:
         type: string
         required: false
         default: ''
+      run-unit-tests:
```
Comment (mikealfare) [inline .github/workflows/release_prep_hatch.yml:56]: Allow for optional unit tests; `dbt-tests-adapter` has no tests. This should also probably be updated to something like `unit-test-command` (which could be empty), to follow suit with `build-command` and `check-command`.

File: .github/workflows/release_prep_hatch.yml [hunk 5]
```diff
@@ -61,6 +64,11 @@ on:
         type: string
         required: false
         default: main
+      working-dir:
```
Comment (mikealfare) [inline .github/workflows/release_prep_hatch.yml:67]: Allow for working directory to be passed in so that `hatch` can reference the correct `pyproject.toml`.

File: dbt-tests-adapter/pyproject.toml [hunk 6]
```diff
@@ -31,14 +31,6 @@ dependencies = [
     "dbt-adapters",
     "pyyaml",
 ]
-
-[project.optional-dependencies]
```
Comment (mikealfare) [inline dbt-tests-adapter/pyproject.toml:35]: This was moved down into the `hatch` environment dependencies elsewhere. This aligns with that decision.

File: dbt-tests-adapter/pyproject.toml [hunk 7]
```diff
@@ -50,6 +42,9 @@ Changelog = "https://github.com/dbt-labs/dbt-adapters/blob/main/CHANGELOG.md"
 requires = ["hatchling"]
 build-backend = "hatchling.build"
 
+[tool.hatch.version]
```
Comment (mikealfare) [inline dbt-tests-adapter/pyproject.toml:45]: This got moved from troubleshooting, but it's position now aligns with the root `pyproject.toml`, so it feels right to keep it here.

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -73,15 +73,29 @@ defaults:
 
 jobs:
   bump-version-generate-changelog:
-    name: "Bump package version, Generate changelog"
-    uses: dbt-labs/dbt-adapters/.github/workflows/release_prep_hatch.yml@main
+    name: "Bump package version, Generate changelog - dbt-adapters"
+    if: ${{ inputs.package == 'dbt-adapters' }}
+    uses: dbt-labs/dbt-adapters/.github/workflows/release_prep_hatch.yml@fix-sdist0-build
```
Comment (mikealfare) [inline .github/workflows/release.yml:78]: Point this back to main before merging

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -73,15 +73,29 @@ defaults:
...
       deploy_to: ${{ inputs.deploy-to }}
       nightly_release: ${{ inputs.nightly_release }}
       target_branch: ${{ inputs.target_branch }}
+      run-unit-tests: true
     secrets: inherit
 
+  bump-version-generate-changelog-dbt-tests-adapter:
+    name: "Bump package version, Generate changelog - dbt-tests-adapter"
+    if: ${{ inputs.package == 'dbt-tests-adapter' }}
+    uses: dbt-labs/dbt-adapters/.github/workflows/release_prep_hatch.yml@fix-sdist0-build
```
Comment (mikealfare) [inline .github/workflows/release.yml:90]: Point this back to main before merging

File: .github/workflows/release.yml [hunk 10]
```diff
@@ -73,15 +73,29 @@ defaults:
 
 jobs:
   bump-version-generate-changelog:
-    name: "Bump package version, Generate changelog"
-    uses: dbt-labs/dbt-adapters/.github/workflows/release_prep_hatch.yml@main
+    name: "Bump package version, Generate changelog - dbt-adapters"
+    if: ${{ inputs.package == 'dbt-adapters' }}
+    uses: dbt-labs/dbt-adapters/.github/workflows/release_prep_hatch.yml@fix-sdist-build
```
Comment (mikealfare) [inline .github/workflows/release.yml:78]: Point back to main before merging.

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -73,15 +73,29 @@ defaults:
...
       deploy_to: ${{ inputs.deploy-to }}
       nightly_release: ${{ inputs.nightly_release }}
       target_branch: ${{ inputs.target_branch }}
+      run-unit-tests: true
     secrets: inherit
 
+  bump-version-generate-changelog-dbt-tests-adapter:
+    name: "Bump package version, Generate changelog - dbt-tests-adapter"
+    if: ${{ inputs.package == 'dbt-tests-adapter' }}
+    uses: dbt-labs/dbt-adapters/.github/workflows/release_prep_hatch.yml@fix-sdist-build
```
Comment (mikealfare) [inline .github/workflows/release.yml:90]: Point back to main before merging.

**Comment by mikealfare** [comment]
This was resolved by https://github.com/dbt-labs/dbt-adapters/pull/118
